### ISSUE TITLE
Encoding issues with python3

### ISSUE THREAD

**Issue by mwindower** [fi-ts/docker-make#1]
With the new docker-make versions there is a known encoding bug, that occurs for some Dockerfiles:

```
Error: 'ascii' codec can't encode character '\u2192' in position 80: ordinal not in range(128)
```

CI-Log: https://github.com/metal-stack/metal-images/runs/924912735

**Comment by Gerrit91** [comment]
Either with the 0.3.3 binary from releases or from source I can build the Docker image from the commit of this Github action. It's pretty hard to reproduce the issue. :cry:

**Comment by mwindower** [comment]
a few lines above the mentioned error, there is this warning:

```bash
subprocess.py:844: RuntimeWarning: line buffering (buffering=1) isn't supported in binary mode, the default buffer size will be used
```

could you try this with Ubuntu 19.10 or 20.04?

I got this on our CI runners that are still 19.10.

**Comment by Gerrit91** [comment]
The subprocess line is just a warning, but I resolved this amongst some other nasty things in this branch: https://github.com/fi-ts/docker-make/pull/2

It is actually possible to build an Alpine compatible binary of docker-make using a Dockerfile like this:

```
# this is in the docker-make PR branch root dir
FROM inn0kenty/pyinstaller-alpine:3.8 as builder
WORKDIR /app
COPY . .
RUN pip install .
RUN PATH="/pyinstaller:$PATH" pyinstaller docker-make.spec

FROM docker
RUN apk update && apk add git
COPY --from=builder /app/dist/docker-make /docker-make
ENTRYPOINT ["/docker-make"]
```

With the resulting image I can build the Dockerfile from metal-images (did not test before and after actions though) on my notebook:

```
docker run --rm -it -v $(pwd):/workdir -v /var/run/docker.sock:/var/run/docker.sock docker-make:test -w /workdir -L --log-level DEBUG --no-push --no-cache 
```

Maybe we can publish this docker-make docker image on Docker Hub and use this as a base image for the Github action. It should also initialize much faster because you would not need to install Docker in the docker-make-action and things like that.

**Comment by Gerrit91** [comment]
Running integration tests from within the docker-make container is actually not so good, I think (we do this for worker images). We should not need to depend on so many prerequisites in the docker-make action container. We should just use it for building the image.
Testing can then be done on the runner itself in a separate action.

**Comment by Gerrit91** [comment]
Should be resolved with #2 and #3.
